app/scripts/modules/ecs/src/serverGroup/configure/serverGroupConfiguration.service.ts

```diff
@@ -524,12 +534,12 @@ export class EcsServerGroupConfigurationService {
 
   public configureVpcId(command: IEcsServerGroupCommand): IEcsServerGroupCommandResult {
     const result: IEcsServerGroupCommandResult = { dirty: {} };
-    if (!command.subnetType) {
+    if (command.subnetType == null && (command.subnetTypes == null || command.subnetTypes.length == 0)) {
       command.vpcId = null;
       result.dirty.vpcId = true;
     } else {
       const subnet = find<ISubnet>(command.backingData.subnets, {
-        purpose: command.subnetType,
+        purpose: command.subnetTypes ? command.subnetTypes[0] : command.subnetType,
```

Comment:
same question as above re: filtering.

Reply:
Users can only deploy to one VPC at a time (multiple subnets that exist in the same VPC). In this case, all of the subnets will be in the same VPC, so it shouldn't matter which subnet you take, since you're using it to get the vpc id.